suitcase/mongo_normalized/__init__.py

```diff
@@ -90,6 +92,24 @@ def __call__(self, name, doc):
         sanitized_doc = event_model.sanitize_doc(doc)
         return super().__call__(name, sanitized_doc)
 
+    def update(self, name, doc):
+        if name in ['start', 'stop']:
+            current_col = getattr(self, f'_run_{name}_collection')
+            revisions_col = getattr(self, f'_run_{name}_collection_revisions')
+            old = current_col.find_one({'uid': doc['uid']})
+            old.pop('_id')
+            try:
```

Comment:
The goal of this code is to catch `StopIteration` raised by `next(cur)`. If something _else_ raises `StopIteration`---perhaps some code deep inside `.find(...)` or `.sort(...)` could do this due to some unforeseen mistake---we will get very confusing behavior. It is safer to put only the specific code we are targeting inside the `try:` block, as in:

```py
try:
    latest_revision = next(cur)
except StopIteration:
    old['revision'] = latest_revision['revision']
    ...
```

See what I mean?

Reply:
You mean this, right? in your code, `latest_revision` wasn't defined in except.
```
target_uid_docs = revisions_col.find({'uid': doc['uid']})
cur = target_uid_docs.sort([('revision', -1)]).limit(1)
try:
    old['revision'] = next(cur)['revision'] + 1 
except StopIteration:
    old['revision'] = 0 
```

Reply:
Indeed, my sketch wasn't quite correct. I had in mind something like:
```py
try:
    latest_revision = next(cur)
except StopIteration:
    old['revision'] = 0
else:
    old['revision'] = latest_revision['revision']
    ...
```
but I think your version is sufficiently safe.